bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -2150,6 +2166,19 @@ private static boolean checkIsLikeOnValue(List<String> errors, Object sourceValu
                     varName);
         }
 
+        if (sourceTypeTag == TypeTags.TYPE_REFERENCED_TYPE_TAG) {
```

Comment:
Lets have separate issue to improve this logic with switch case

Reply:
created https://github.com/ballerina-platform/ballerina-lang/issues/36443